packages/ld-key-pair/KeyPair.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * defines the ability to sign based on a KeyPair or only
+ * from a private key
+ */
+interface Signer {
+    privateKey: string;
+    keyPair?: KeyPair;
+    sign(options: {
+        data: any;
+    }): string;
+}
+
+/**
+ * defines the ability to perform verification based on a KeyPair
+ * or a public key
+ */
+interface Verifier {
+    publicKey: string;
+    keyPair?: KeyPair;
+    verify(options: {
+        data: any;
+        signature: string;
+    }): boolean;
+}
+
+/**
+ * public key side of a key pair
+ */
+interface PublicNode {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    publicKeyBase58?: string;
+}
+/**
+ * public key side of a key pair
+ */
+ interface PrivateNode {
+    id: string;
+    type: any;
+    controller?: any;
+    privateKey?: string;
+    privateKeyBase58?: string;
+}
+
+/**
+ * defines the common properties and methoods in use by 
+ * key pairs in the LD realm based on existing implementations
+ */
+interface KeyPair {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    privateKey?: string;
+    publicKeyBase58?: string;
+    privateKeyBase58?: string;
+    publicNode(): PublicNode;
+    generate(options: any): KeyPair;
+    fingerprint(): string;
+    verifyFingerprint(options: {
+        fingerprint: string
+    }): boolean;
+    fromFingerprint(options: {
+        fingerprint: string
+    }): KeyPair;
+    fingerprintFromPublicKey(options: {
+        publicKeyBase58: string
+    }): string;
+    signer(): Signer;
+    verifier(): Verifier;
+}
+
+/**
+ * allows forthe general grouping of a list of key pairs that are
+ * related in some fashion
+ */
+interface KeyPairs {
+    id: string;
+    type: any;
+    controller?: any;
+    pairs?: KeyPair[];
```

Comment:
concerned about what this could really mean...

an unbounded array of public keys, private keys, public and private?

There is a semantics problem with using the word "KeyPair" to describe both public and private keys.

in a way, private keys are the only "key pair" since public keys don't have a private component.

Not a hill I would die on, just commenting on the issue.

I think this ship has officially sailed... https://developer.mozilla.org/en-US/docs/Web/API/CryptoKeyPair